sourced/ml/transformers/uast2bag_features.py

```diff
@@ -28,26 +28,34 @@ def documentize(self, r: Row) -> Row:
         return Row(**{bfc.document: doc, ec.Uast: r[ec.Uast]})
 
 
-class Uast2BagFeatures(Transformer):
-    class Columns:
-        """
-        Standard column names for interop.
-        """
-        token = "token"
-        document = "document"
-        value = "value"
-
-    def __init__(self, extractors: Iterable[BagsExtractor], **kwargs):
+class Uast2Features(Transformer):
+    def __init__(self, *extractors: Extractor, **kwargs):
         super().__init__(**kwargs)
         self.extractors = extractors
 
     def __call__(self, rows: RDD):
         return rows.flatMap(self.process_row)
 
+    def process_feature(self, row: Row, name, feature):
```

Comment:
This must be implemented in the derived class.

UastMiner -> Uast2FeaturesDict, Uast2BagOfFeatures

(I am open to better names)

Reply:
UastMiner, Uast2Features, Uast2BagFeatures?